app/javascript/hotwire/spark/reloaders/html_reloader.js

```diff
@@ -2,19 +2,37 @@ import { Idiomorph } from "idiomorph/dist/idiomorph.esm.js"
 import { log } from "../logger.js"
 import { reloadHtmlDocument } from "../helpers.js"
 import { StimulusReloader } from "./stimulus_reloader.js"
+import HotwireSpark from "../index.js"
 
 export class HtmlReloader {
   static async reload() {
     return new HtmlReloader().reload()
   }
 
   async reload() {
-    const reloadedDocument = await this.#reloadHtml()
-    await this.#reloadStimulus(reloadedDocument)
+    if(HotwireSpark.config.htmlReloadStrategy == "morph") {
```

Comment:
I think the place to make this call is in monitoring channel: it will configure one reloader or the other depending on the config options. That way, we have separated reloaders for each option. They are quite different from each other, so I think it's better if each reload method has its own class:

```js
  reloadHtml() {
    return this.#htmlReloader.reload()
  },

  get #htmlReloader() {
    if (HotwireSpark.config.htmlReloadMethod == "morph") {
      return MorphHtmlReloader
    } else {
      return ReplaceHtmlReloader
    }
  }
```